### PR TITLE
Introduce WithWindowsBrokerOptions and use AAD only interactive experience if possible

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/ApiConfig/WindowsBrokerOptions.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+{
+
+    /// <summary>
+    /// Advanced options for using the Windows 10 broker.
+    /// For more details see https://aka.ms/msal-net-wam
+    /// </summary>
+#if !SUPPORTS_BROKER || __MOBILE__
+    [EditorBrowsable(EditorBrowsableState.Never)]
+#endif    
+    public class WindowsBrokerOptions
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/ApiConfig/WindowsBrokerOptions.cs:23]: @jmprieur  to review public API

File: src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs [hunk 2]
```diff
@@ -176,6 +177,19 @@ public PublicClientApplicationBuilder WithBroker(bool enableBroker = true)
 #pragma warning restore CS0162 // Unreachable code detected
         }
 
+        /// <summary>
+        /// Allows customization of the Windows 10 Broker experience
+        /// </summary>
+#if !SUPPORTS_BROKER || __MOBILE__
+        [EditorBrowsable(EditorBrowsableState.Never)]
+#endif  
+        public PublicClientApplicationBuilder WithWindowsBrokerOptions(WindowsBrokerOptions options)
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:186]: @jmprieur  to review public API

File: src/client/Microsoft.Identity.Client/ApiConfig/WindowsBrokerOptions.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+using Microsoft.Identity.Client.PlatformsCommon.Factories;
+
+namespace Microsoft.Identity.Client
+{
+
+    /// <summary>
+    /// Advanced options for using the Windows 10 broker.
+    /// For more details see https://aka.ms/msal-net-wam
+    /// </summary>
+#if !SUPPORTS_BROKER || __MOBILE__
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/ApiConfig/WindowsBrokerOptions.cs:20]: Does `__MOBILE__` flag needs to be added to props?

File: src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WamBroker.cs [hunk 4]
```diff
@@ -73,9 +74,12 @@ internal class WamBroker : IBroker
             _aadPlugin = testAadPlugin ?? new AadPlugin(_wamProxy, _webAccountProviderFactory, _logger);
             _msaPlugin = testmsaPlugin ?? new MsaPlugin(_wamProxy, _webAccountProviderFactory, _logger);
 
-            _msaPassthroughHandler = msaPassthroughHandler ?? 
+            _msaPassthroughHandler = msaPassthroughHandler ??
                 new MsaPassthroughHandler(_logger, _msaPlugin, _wamProxy, _parentHandle);
 
+            _wamOptions = appConfig.WindowsBrokerOptions ?? 
+                new WindowsBrokerOptions();
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WamBroker.cs:81]: nit: What it be appropriate to call `WindowsBrokerOptions.CreateDefault()` here?

File: tests/Microsoft.Identity.Test.Unit/BrokerTests/WamTests.cs [hunk 5]
```diff
@@ -603,6 +649,103 @@ public async Task RemoveAADAccountAsync()
             }
         }
 
+        [TestMethod]
+        public async Task GetAccounts_DoesNotCallPlugins_Async()
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Unit/BrokerTests/WamTests.cs:653]: Should this test method be **CallsPlugins** and next one **DoesNotCallPlugins**?

**Review by pmaytak — APPROVED** [review]

**Review by jmprieur — COMMENTED** [review]
The changes to the public API look good to me

File: src/client/Microsoft.Identity.Client/ApiConfig/WindowsBrokerOptions.cs (hunk 3 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/ApiConfig/WindowsBrokerOptions.cs:20]: No, it's defined by .NET

**Pull request merged**